Sources/Classes/PlayerCore/TinyVideoPlayer.swift

```diff
@@ -338,6 +338,7 @@ public class TinyVideoPlayer: NSObject, TinyVideoPlayerProtocol, TinyLogging {
 
         /* Attach an AVPlayerItem to player immediately triggers the media preparation, KVO works after here. */
         player.replaceCurrentItem(with: playerItem)
+        currentVideoPlaybackEnded = false
```

Comment:
I'm not sure about this change. Have you encountered an issue before adding this line?

Line 340 should enable the internal loading operation and kick-off the KVO stuff. When it's ready, the observer (which is TinyPlayer itself) should receive a `AVPlayerItem.status` value change (line 570) and eventually call the `-playerItemIsReadyForPlaying(:)` method. In this method, the `currentVideoPlaybackEnded` property is already set to `false`. 

This guarantees for each newly loaded `PlayerItem`, before playing this property is always set to `false`.

If you have had an issue before, please share an example code snippet with me. I'm happy to investigate it. Thanks! 😃